lib/web_ui/lib/src/engine/window.dart

```diff
@@ -29,8 +31,17 @@ const bool debugPrintPlatformMessages = false;
 /// The view ID for the implicit flutter view provided by the platform.
 const int kImplicitViewId = 0;
 
+/// Represents all views in the Flutter Web Engine.
+///
+/// In addition to everything defined in [ui.FlutterView], this class adds
+/// a few web-specific properties.
+abstract class EngineFlutterView extends ui.FlutterView {
```

Comment:
My thought here was that `EngineFlutterView` will become the base for all window and non-window views. Other non-window views will directly extend `EngineFlutterView` since they don't need `ui.SingletonFlutterWindow`.

So the hierarchy becomes:
```
                         ui.FlutterView
               (dart:ui interface across platforms)
                                |
                                |
                       EngineFlutterView (new)
           (web-specific additions to the interface)
                /                               \
               /                                 \
     NonWindowFlutterView (future)          EngineFlutterWindow
(non-sintleton, non-window views)        (legacy singleton window)

```

> Also, I think this should be `abstract interface class EngineFlutterView`, if we can use that [feature of Dart](https://dart.dev/language/class-modifiers#interface).

We can! We already use it in `platform_location.dart` :)

Reply:
Ahhh, I gotcha. The advantage of the current version:

`ui.View -> ui.Window -> WebWindow implements WebOnlyAccessors`

Is that we're only extending the base classes defined by the core of the engine. In my mind that means that if there's an upcoming change in the base structure of classes in package:ui, it should be easier for `ui` to do changes to the base class, and for us to adapt to those changes?

(Of course, it depends on the changes, so our class extensions might not matter too much!)